src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -249,179 +256,149 @@ public async Task RefreshAsync(CancellationToken cancellationToken)
+                                _ffEtags,
+                                client,
+                                cancellationToken).ConfigureAwait(false);
+
+                            if (ffCollectionUpdated)
                             {
-                                // Trigger a single load-all operation if a change was detected in one or more key-values with refreshAll: true
-                                data = await LoadSelectedKeyValues(client, cancellationToken).ConfigureAwait(false);
-                                watchedSettings = await LoadKeyValuesRegisteredForRefresh(client, data, cancellationToken).ConfigureAwait(false);
-                                watchedSettings = UpdateWatchedKeyValueCollections(watchedSettings, data);
-                                logInfoBuilder.AppendLine(LogHelper.BuildConfigurationUpdatedMessage());
-                                return;
-                            }
+                                ffEtags = new Dictionary<KeyValueSelector, IEnumerable<MatchConditions>>();
 
-                            changedKeyValuesCollection = await GetRefreshedKeyValueCollections(refreshableMultiKeyWatchers, client, logDebugBuilder, logInfoBuilder, endpoint, cancellationToken).ConfigureAwait(false);
+                                ffCollectionData = await LoadSelected(
+                                    client,
+                                    new Dictionary<KeyValueSelector, IEnumerable<MatchConditions>>(),
+                                    ffEtags,
+                                    _options.Selectors.Where(selector => selector.IsFeatureFlagSelector),
+                                    cancellationToken).ConfigureAwait(false);
 
-                            if (!changedKeyValuesCollection.Any())
+                                logInfoBuilder.Append(LogHelper.BuildFeatureFlagsUpdatedMessage());
+                            }
+                            else
                             {
                                 logDebugBuilder.AppendLine(LogHelper.BuildFeatureFlagsUnchangedMessage(endpoint.ToString()));
                             }
-                        },
-                        cancellationToken)
-                        .ConfigureAwait(false);
+                        }
+                    },
+                    cancellationToken)
+                    .ConfigureAwait(false);
 
-                    if (!refreshAll)
+                    if (refreshAll)
                     {
-                        watchedSettings = new Dictionary<KeyValueIdentifier, ConfigurationSetting>(_watchedSettings);
+                        _mappedData = await MapConfigurationSettings(data).ConfigureAwait(false);
 
-                        foreach (KeyValueWatcher changeWatcher in refreshableWatchers.Concat(refreshableMultiKeyWatchers))
+                        // Invalidate all the cached KeyVault secrets
+                        foreach (IKeyValueAdapter adapter in _options.Adapters)
```

Comment:
Where do deleted flags get removed?

Reply:
discussed offline, lost this functionality along the way when restructuring the refresh methods. I'll try checking the existing order and using that to decide whether the setting for that flag's key should be updated.